### DISCUSSION TITLE
중복 결제 문의

### DISCUSSION THREAD

**Discussion by Guwon** [commerce-api-naver/commerce-solutions#39] [묻고 답하기]
안녕하세요 유니드컴즈 유동진입니다.

다음 사용자로부터 중복 결제 문의가 인입되어 확인 요청드립니다.
해당 이슈는 커머스 고객센터로 안내드렸으나, 다시 저희 쪽으로 문의가 들어왔는데 해당 채널로 문의드려도 괜찮을지요?

- account_uid :ncp_2xZQw6vvKypinIvamMy7k
- solution_id : 62jmw6IBNQ9w3Qi0RfA48Y
- event_id : 3LLABXJqgYHQOhXR0y4MKK, 18QGHG8ipth4O4E8RUqUoq

![스크린샷 2024-10-23 오전 10 49 42](https://github.com/user-attachments/assets/11305978-3342-4835-8310-cf97ade8d8ed)

![스크린샷 2024-10-23 오전 10 49 47](https://github.com/user-attachments/assets/8522f1fb-55fe-48e0-8e19-034b857de480)

![스크린샷 2024-10-23 오후 5 01 59](https://github.com/user-attachments/assets/46176272-ec55-4818-a4af-baed336fedcb)

**Comment by commerce-api-naver** [comment]
안녕하세요. 커머스API 이민석입니다.

## 문제 원인
대상 사용자 구독건에 대하여 '[솔루션] 사용 시작 승인 (`PUT /v1/commerce-solutions/subscriptions/approve`)' API가 아래 시간대 2회 호출된 것으로 확인됩니다.
- 2024-10-22T16:05:44.604 KST
- 2024-10-22T16:05:45.731 KST

### 개선 예정 내용
'심사 후 승인'형 솔루션의 사용 승인 처리와 관련하여 짧은 시간대 다량의 호출이 발생할 경우 이를 처리하는 로직에서의 문제점을 확인하였습니다.
해당 내용은 최대한 빠르게 수정하여 11월 중 개선 내용을 배포할 예정입니다.
솔루션 운용에 불편을 드리어 죄송합니다.

## 문의주신 건에 대한 확인 사항
1. 대상 구독건에 대한 전체 강제 해지 및 적절한 금액으로 환불 처리 후 재구독으로 안내가 가능하실까요?
유니드컴즈 측에서 환불이 어려우신 경우 알려주시면 저희쪽에서 해지 대행을 진행해드리도록 하겠습니다.

2. 향후 유사 현상 발생 시 솔루션사를 통해 기술지원 채널로 문의 부탁드립니다.
해당하는 유형은 정확한 원인 확인과 데이터 확인이 필요하여 스마트스토어 고객센터를 통해서는 해결이 어려울 수 있습니다.

**Reply by Guwon** [reply]
@commerce-api-naver 환불은 저희 쪽에서 처리하도록 하겠습니다. 빠른 확인 감사합니다.

**Reply by commerce-api-naver** [reply]
@Guwon 님, 커머스API 이민석입니다.

앞서 말씀드린 것과 같이 문의주신 현상의 시작은 동일 API 내용을 연속하여 호출함으로서 발생한 내용입니다.
커머스솔루션에서도 관련 로직을 보강할 예정이나 유니드컴즈 쪽에서도 특히 솔루션의 구독/승인과 관련한 API들은 구독자의 행동과 관계없이 1회 API 호출로만 처리되도록 설계하시는 것을 권해드립니다.

**Reply by Guwon** [reply]
@commerce-api-naver 넵 저희 쪽에서도 관련 로직을 보강할 예정입니다. 전달 주셔서 감사합니다!

**Comment by commerce-api-naver** [comment]
@Guwon 님 안녕하세요. 커머스API 이민석입니다.

문의주신 건과 관련한 내부적인 대응 변경을 검토하는 중 과거 유사한 사례로 추정되는 케이스를 더 발견하여 알려드립니다.
대상 건들은 스마트스토어 고객센터 채널을 통한 문의는 없었던 건들인지라 유니드컴즈 측에서 필요 시 아래 정보로 대응을 진행해주시면 될 것 같습니다.
현재 대상건은 아래 3건으로 대상 회차가 종료되었거나 해지가 완료된 상태입니다. 따라서 네이버페이 비즈월렛 이외의 경로로 환불 검토가 필요할 수 있을 것으로 보입니다.

혹시 대상 결제건 확인이 추가 정보가 필요하실 경우 말씀 부탁드리겠습니다.

### 추가 케이스
#### 케이스 1
```plain
"solutionId" : "238XF5ftP4l2spJdI3RALR",
"accountUid" : "ncp_2sSF1ZTW0SdAoVL7Y6YXe",
"subscriptionId" : "2sWI0ELTmhHrtKYrFUptAd",
"totalAmount" : "19900",
"date" : "2024-09-25 04:14"
```

#### 케이스 2
```plain
"solutionId" : "238XF5ftP4l2spJdI3RALR",
"accountUid" : "ncp_2sSF1SUtetKDEkXw7V3F5", 
"subscriptionId" : "1ccMldH7oMcbtkqJQW9mnJ",
"totalAmount" : "19900",
"date" : "2024-09-19 07:34"
```

#### 케이스 3
```plain
"solutionId" : "62jmw6IBNQ9w3Qi0RfA48Y",
"accountUid" : "ncp_2sSEy1ZGYMeUm0iZVRFfd",
"subscriptionId" : "7TpzcDxzhpq4bbv8Qcj8Sf",
"totalAmount" : "19900",
"date" : "2024-08-21 00:59"
```

**Reply by Guwon** [reply]
@commerce-api-naver 넵 추가 정보 필요시 요청드리겠습니다. 확인 감사합니다.
